File: backend/models/Connection.js
const { DataTypes } = require('sequelize');
const { sequelize } = require('../db');

const Connection = sequelize.define('Connection', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true
  },
  source: {
    type: DataTypes.UUID,
    allowNull: false,
    references: { model: 'notes', key: 'id' }
  },
  target: {
    type: DataTypes.UUID,
    allowNull: false,
    references: { model: 'notes', key: 'id' }
  },
  sourceHandle: {
    type: DataTypes.STRING,
    allowNull: true
  },
  targetHandle: {
    type: DataTypes.STRING,
    allowNull: true
  },
  label: {
    type: DataTypes.STRING,
    defaultValue: ''
  }
}, {
  tableName: 'connections',
  timestamps: true,
  createdAt: 'createdAt',
  updatedAt: false
});

Connection.prototype.toJSON = function () {
  const values = { ...this.get() };
  values._id = values.id;
  return values;
};

module.exports = Connection;

File: backend/app/models/connection.py
import uuid
from datetime import datetime
from typing import Optional
from uuid import UUID

from sqlalchemy import DateTime, ForeignKey, String, func
from sqlalchemy.dialects.postgresql import UUID as PG_UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.database import Base


class Connection(Base):
    __tablename__ = "connections"

    id: Mapped[UUID] = mapped_column(
        PG_UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    source: Mapped[UUID] = mapped_column(
        PG_UUID(as_uuid=True), ForeignKey("notes.id"), nullable=False
    )
    target: Mapped[UUID] = mapped_column(
        PG_UUID(as_uuid=True), ForeignKey("notes.id"), nullable=False
    )
    source_handle: Mapped[Optional[str]] = mapped_column("sourceHandle", String, nullable=True)
    target_handle: Mapped[Optional[str]] = mapped_column("targetHandle", String, nullable=True)
    label: Mapped[str] = mapped_column(String, default="")
    created_at: Mapped[datetime] = mapped_column(
        "createdAt", DateTime(timezone=True), default=func.now(), server_default=func.now()
    )